packages/carequality-sdk/src/client/carequality-fhir.ts

```diff
@@ -0,0 +1,201 @@
+  private static readonly stagingUrl = "https://directory.stage.carequality.org/fhir";
+  private static readonly productionUrl = "https://directory.prod.carequality.org/fhir";
+
+  static ORG_ENDPOINT = "/Organization";
+  readonly api: AxiosInstance;
+  readonly apiKey: string;
+  readonly maxBackoff: number;
+  private httpsAgent: Agent;
+
+  /**
+   * Creates a new instance of the Carequality Management API client pertaining to an
+   * organization to make requests on behalf of.
+   *
+   * @param orgCert                 The certificate (public key) for the organization.
+   * @param rsaPrivateKey           An RSA key corresponding to the specified orgCert.
+   * @param rsaPrivateKeyPassword   The password to decrypt the private key.
+   * @param apiKey                  The API key to use for authentication.
+   * @param apiMode                 Optional, the mode the client will be running. Defaults to staging.
+   * @param options                 Optional parameters
+   * @param options.timeout         Connection timeout in milliseconds, defaults to 120 seconds.
+   * @param options.retries         Number of retries for failed requests, defaults to 3.
+   * @param options.maxBackoff      Number of seconds for the maximum backoff during retry requests, defaults to 30 seconds.
+   */
+  constructor({
+    orgCert,
+    rsaPrivateKey,
+    rsaPrivateKeyPassword,
+    apiKey,
+    apiMode = APIMode.production,
+    options = {},
+  }: {
+    orgCert: string;
+    rsaPrivateKey: string;
+    rsaPrivateKeyPassword: string;
+    apiKey: string;
+    apiMode: APIMode;
+    options?: { timeout?: number; retries?: number; maxBackoff?: number };
+  }) {
+    this.httpsAgent = new Agent({
+      cert: orgCert,
+      key: rsaPrivateKey,
+      passphrase: rsaPrivateKeyPassword,
+    });
+    let baseUrl;
+
+    switch (apiMode) {
+      case APIMode.dev:
+        baseUrl = CarequalityManagementApiFhir.devUrl;
+        break;
+      case APIMode.staging:
+        baseUrl = CarequalityManagementApiFhir.stagingUrl;
+        break;
+      case APIMode.production:
+        baseUrl = CarequalityManagementApiFhir.productionUrl;
+        break;
+      default:
+        throw new Error("API mode not supported.");
+    }
+    this.maxBackoff = options?.maxBackoff
+      ? dayjs.duration(options.maxBackoff, "seconds").asMilliseconds()
+      : DEFAULT_MAXIMUM_BACKOFF.asMilliseconds();
+
+    this.api = axios.create({
+      timeout: options?.timeout ?? DEFAULT_AXIOS_TIMEOUT.asMilliseconds(),
+      baseURL: baseUrl,
+      httpsAgent: this.httpsAgent,
+    });
+
+    // TODO: #1536 - improved retry logic. Issue: https://github.com/metriport/metriport-internal/issues/1536
+    axiosRetry(this.api, {
+      retries: options?.retries ?? DEFAULT_MAX_RETRIES,
+      retryDelay: retryCount => {
+        const exponentialDelay = Math.pow(2, Math.max(0, retryCount - 1));
+        const jitter = Math.random();
```

Comment:
whats the jitter for?

Reply:
To minimize the same delay between consecutive calls. That attempts to spread out the load on the server.

Reply:
cool!